Source/Urho3D/AngelScript/Script.cpp

```diff
@@ -311,6 +311,46 @@ asITypeInfo* Script::GetObjectType(const char* declaration)
     return type;
 }
 
+const char **Script::GetEnumValues(int asTypeID)
+{
+    // If we've already found it, do not try to update the values, as that may invalidate the buffer
+    if (enumValues_.Contains(asTypeID))
+        return enumValues_[asTypeID].Buffer();
+
+    asITypeInfo* type = scriptEngine_->GetTypeInfoById(asTypeID);
+    if (!type)
+        return nullptr;
+
+    if (type->GetFlags() & asOBJ_ENUM)
+    {
+        unsigned count = type->GetEnumValueCount();
+        enumValues_[asTypeID].Resize(count + 1);
+        int val;

```

Comment:
Please move it inside the loop. And uninitialized variables sux, yeah.

Reply:
You are just referring to val, right?
Also, if I may ask, why are you so opposed to uninitialized variables? I don't think you're wrong, I just want to know why, especially in this case, where the functions fill them with values.

Reply:
> You are just referring to val, right?

Yep

> Also, if I may ask, why are you so opposed to uninitialized variables?

They just provoke shit to happen... Just like in the issue above.
Maybe one add new variable and leave it uninitialized too.
Maybe one change function so it won't overwrite the variable _sometimes_.
When you always initialize everything, it makes code more errorproof and stable.